incubator/kafka/templates/statefulset.yaml

```diff
@@ -85,16 +115,17 @@ spec:
           name: external-{{ $i }}
           {{- end }}
         {{- end }}
-        {{- if .Values.jmxPort }}
-        - containerPort: {{ .Values.jmxPort }}
+        {{- if .Values.metrics.jmx.enabled }}
```

Comment:
After merging with master, this conflict popped up. Not sure the best way to deal with it; I'm a bit disappointed that the `.Values.jmxPort` has set the standard, as it will be difficult to change now. @ethanocentricity this was your PR; do you have an opinion?

Reply:
If you want to switch it around, I don't have a problem.  My goal was to get jmx functionality quickly (for tools like kafka manager and prometheus).
Yours is more complete since you manage the prom yaml for the exporter, so that's wonderful.

It seems like you are doing it with a separate docker container for the prom exporter, which is fine with me.  I was using the in process java agent per https://www.robustperception.io/monitoring-kafka-with-prometheus/ when I used this on my k8s.   
But I've exposed metrics both ways and don't really have any preference.

So, assuming your commit supports both jmx scraping from external components and exposing prom metrics, feel free to change whatever you like.

Reply:
@nrmitchi as @ethanocentricity said this can be moved around. This chart is still in incubation so we may make changes that require some attention upon upgrades (we should definitely do a minor bump with this chart). Since you've built out metrics in this PR feel free to move this under your `metrics` key